src/modules/accounting/accounting.py

```diff
@@ -393,9 +460,176 @@ def _combine_report_parts(
             el_rewards_vault_balance=el_rewards_vault_balance,
             shares_requested_to_burn=shares_requested_to_burn,
             withdrawal_finalization_batches=finalization_batches,
-            finalization_share_rate=finalization_share_rate,
             is_bunker=is_bunker,
+            tree_root=tree_root,
+            tree_cid=tree_cid,
             extra_data_format=extra_data.format,
             extra_data_hash=extra_data.data_hash,
             extra_data_items_count=extra_data.items_count,
         )
+
+    @staticmethod
+    def _get_ether_by_shares(shares: int, pre_total_ether: int, pre_total_shares: int) -> float:
+        return (shares * pre_total_ether) / pre_total_shares
+
+    # pylint: disable=too-many-branches,too-many-statements
+    def _get_vaults_fees(self, blockstamp: ReferenceBlockStamp, vaults: VaultsMap, vaults_total_values: list[int]) -> list[int]:
+        prev_report = self.w3.staking_vaults.get_prev_report(blockstamp)
+        ## When we do not have a report - then we do not have starting point for calculation
+        if prev_report is None:
+            rebased_event = self.w3.lido_contracts.lido.get_last_token_rebased_event(blockstamp.block_number - 7200, blockstamp.block_number)
```

Comment:
we shall not rely on magic 7200 number and rather retrieve the latest delivered report

https://docs.lido.fi/contracts/accounting-oracle#getlastprocessingrefslot